build/index.js

```diff
@@ -195,7 +195,27 @@ async function buildDocument(document, documentOptions = {}) {
     if (options.clearKumascriptRenderCache) {
       renderKumascriptCache.clear();
     }
-    [renderedHtml, flaws] = await kumascript.render(document.url);
+    try {
+      [renderedHtml, flaws] = await kumascript.render(document.url);
+    } catch (error) {
+      if (error.name === "MacroInvocationError") {
+        // The source HTML couldn't even be parsed! There's no point allowing
+        // anything else move on.
+        // But considering that this might just be one of many documents you're
+        // building, let's at least help by setting a more user-friendly error
+        // message.
+        throw new Error(
+          `MacroInvocationError trying to parse ${
+            document.fileInfo.path
+          }, line ${error.line + document.fileInfo.frontMatterOffset} column ${
+            error.column
+          } (${error.error.message})`
+        );
```

Comment:
The line number is off by 1. There are two options here. First, you could take advantage of the `updateFileInfo` method of the base error class, so that would look like this:
```suggestion
        error.updateFileInfo(document.fileInfo);
        throw new Error(
          `MacroInvocationError trying to parse ${error.filepath}, line ${error.line} column ${error.column} (${error.error.message})`
        );
```

or just subtract 1 from the line caculation, so something like this:

```suggestion
        throw new Error(
          `MacroInvocationError trying to parse ${
            document.fileInfo.path
          }, line ${error.line + document.fileInfo.frontMatterOffset - 1} column ${
            error.column
          } (${error.error.message})`
        );
```